### PR TITLE
Fix(Auth): Refactor devicehelper to keep what is required for cryptoBR

### PR DESCRIPTION
**Pull request by gpanshu**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

To ensure we do not add code that is not required, this refactor was necessary and removes redundant code.

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [x] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/SRPHelper.kt [hunk 1]
```diff
@@ -39,23 +39,23 @@ class SRPHelper(private val password: String) {
...
-    // Generator 'g' parameter.
-    private val g = BigInteger.valueOf(2)
-
-    // Precomputed safe 3072-bit prime 'N', as decimal.
-    // https://datatracker.ietf.org/doc/html/rfc5054#appendix-A (Page 16)
-    private val HEX_N =
-        "FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B139B22514A08798E3404DDEF9519B3CD3A" +
+        @JvmStatic()
+        fun getHexN(): String = "FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B139B22514" +
+            "A08798E3404DDEF9519B3CD3A" +
```
Comment (tylerjroach) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/SRPHelper.kt:45]: odd early line break but not important

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelper.java [hunk 2]
```diff
@@ -80,7 +67,7 @@ public static String generateRandomString() {
      * Static class for SRP related calculations for devices.
      */
     @SuppressWarnings("checkstyle:typename")
-    public static class deviceSRP {
+    private static final class deviceSRP {
```
Comment (tylerjroach) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelper.java:70]: Why are we using lowercase for a class?

**Review by tylerjroach — APPROVED** [review]

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelperTests.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.amplifyframework.auth.cognito.helpers
+
+import kotlin.test.assertEquals
+import org.junit.Test
+
+class CognitoDeviceHelperTests {
+
+    @Test
+    fun testHex() {
```
Comment (sktimalsina) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelperTests.kt:9]: nit: is `test` prefix required? Previous ver of JUnit needed it. But I don't think it is necessary anymore.
How about naming this method as in what it tests
generated_hex_is_correct or sth like that.

**Review by sktimalsina — APPROVED** [review]
few comments, nothing blocking

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/SRPHelper.kt (hunk 1 above)
Comment (gpanshu) [reply aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/SRPHelper.kt:45]: Yeah I was being lazy. Fixed.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelper.java (hunk 2 above)
Comment (gpanshu) [reply aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelper.java:70]: I tried to minimize my impact. I will change this.

**Pull request merged**